README.md

```diff
@@ -2839,6 +2839,29 @@ function quux(foo, bar, baz) {
 }
 // Settings: {"jsdoc":{"preferredTypes":{"hertype":{"replacement":"HerType<>"},"histype":"HisType.<>"}}}
 // Options: [{"definedTypes":["MyType"],"preferredTypesDefined":true}]
+
+/**
+ * @template TEMPLATE_TYPE
+ */
+class Foo {
+  /**
+   * @return {TEMPLATE_TYPE}
+   */
+  bar () {
+  }
+}
+
+/**
+ * @template TEMPLATE_TYPE_A, TEMPLATE_TYPE_B
+ */
+class Foo {
+  /**
+   * @param {TEMPLATE_TYPE_A} baz
+   * @return {TEMPLATE_TYPE_B}
+   */
+  bar (baz) {
```

Comment:
Please add an invalid example corresponding to this one as well, though without `@template` (and with the appropriate error).

Reply:
I originally thought this might be redundant since all invalid types can essentially be treated as generic types with missing `@template` tag and thus covered by existing tests.

Still, I added such a test case and also one that check that generic types are scoped correctly to their class and not treated as globals as a potential invalid implementation could do.